ext/extconf.rb

```diff
@@ -27,7 +27,7 @@
 # darwin nix clang doesn't support lto
 # disable -lto flag for darwin + nix
 # see: https://github.com/sass/sassc-ruby/issues/148
-enable_lto_by_default = (Gem::Platform.local.os == "darwin" && !ENV['NIX_CC'].nil?)
+enable_lto_by_default = (Gem::Platform.local.os == "darwin" && ENV['NIX_CC'].nil?)
```

Comment:
On closer inspection this check is a little weird. Firstly I'm pretty sure `nil?` is a Rails addition. Secondly this should enable LTO by default _unless_ we are using nix clang on darwin. Maybe:

```
enable_lto_by_default = !(Gem::Platform.local.os == "darwin" && ENV['NIX_CC'])
```

Reply:
Thanks for having a look at this :).

I _think_ `.nil?` "comes standard".

```
~ $ irb

WARNING: This version of ruby is included in macOS for compatibility with legacy software.
In future versions of macOS the ruby runtime will not be available by
default, and may require you to install an additional package.

irb(main):001:0> nil.nil?
=> true
irb(main):002:0> "".nil?
=> false
irb(main):003:0>
```

I agree the check is weird and I think your version makes more sense. It reads well: "Enable LTO by default when we're not on Darwin _and_ the nix wrapped C compiler will be used".

Reply:
Oh sorry about that, you're absolutely right!

Also just to make it more explicit, it looks like in `master` the only time LTO is automatically enabled is on Darwin with Nix clang, which is the opposite to what we want.

Reply:
Yep, sorry. The title of my branch / PR is weird :P. In my head it was "Fix check to disable the enabling of LTO by default" ... ergh. 

I wonder if there isn't a better way to do this in general. Like to detect if LTO will work without having to _know_ that it won't work for the C compiler wrapped by Nix on Darwin. Like, could it also not work with other C compilers? Is there some way to ask: "is LTO available?". 

Until then it'd be good to get something like this merged as I'm sure other Darwin + Nix using Ruby devs are having a head-scratching time.